### DISCUSSION TITLE
High cpy usage compared to mavlink-routerd

### DISCUSSION THREAD

**Discussion by patrickelectric** [bluenviron/mavp2p#44] [Q&A]
I'm running mavp2p with the following cli:
`/usr/bin/mavp2p udps:127.0.0.1:8852 udpc:127.0.0.1:14000 udpc:192.168.2.1:14550 udps:0.0.0.0:14550 tcps:127.0.0.1:5777 udps:0.0.0.0:14660 --streamreq-frequency=10`

The idle usage is around 10% and when a GCS is connected goes around 18% in a raspberry pi 4.
The same environment with mavlink-routerd goes between 1.5% to max 3.2%.

Any tips to improve the cpu usage ?

**Comment by aler9** [comment]
Hello, on Raspberry Pi 4, using the command line you provided, usage when idle is 0%, and when a GCS is connected, usage is below 2%:

![Screenshot_20230922_134130](https://github.com/bluenviron/mavp2p/assets/46489434/72fddfeb-7b25-4430-b560-4f4a186385d6)

If you still have problems, provide additional informations in order to replicate the issue.
